libs/pika/async_mpi/src/mpi_polling.cpp

```diff
@@ -742,85 +748,87 @@ namespace pika::mpi::experimental {
+                if (pool_mode == pool_create_mode::pika_decides)
+                {
+                    // required = MPI_THREAD_SINGLE;
+                    // PIKA_DETAIL_DP(detail::mpi_debug<0>, debug(str<>("MPI_THREAD_SINGLE")));
+                }
             }
+            pika::util::mpi_environment::init(nullptr, nullptr, required, required, provided);
             MPI_Comm_rank(MPI_COMM_WORLD, &detail::mpi_data_.rank_);
             MPI_Comm_size(MPI_COMM_WORLD, &detail::mpi_data_.size_);
         }
         else
         {
-            // Check if MPI_Init has been called previously
-            if (detail::mpi_data_.size_ == -1)
+            int is_initialized = 0;
+            MPI_Initialized(&is_initialized);
+            if (is_initialized)
             {
-                int is_initialized = 0;
-                MPI_Initialized(&is_initialized);
-                if (is_initialized)
-                {
-                    MPI_Comm_rank(MPI_COMM_WORLD, &detail::mpi_data_.rank_);
-                    MPI_Comm_size(MPI_COMM_WORLD, &detail::mpi_data_.size_);
-                }
+                MPI_Comm_rank(MPI_COMM_WORLD, &detail::mpi_data_.rank_);
+                MPI_Comm_size(MPI_COMM_WORLD, &detail::mpi_data_.size_);
             }
+            else { PIKA_DETAIL_DP(detail::mpi_debug<0>, error(str<>("mpi not initialized"))); }
         }
 
         PIKA_DETAIL_DP(
-            detail::mpi_debug<5>, debug(str<>("pika::mpi::experimental::init"), detail::mpi_data_));
+            detail::mpi_debug<0>, debug(str<>("pika::mpi::experimental::init"), detail::mpi_data_));
 
+        // --------------------------------------
+        // install error handler (convert mpi errors into exceptoions
         if (init_errorhandler)
         {
             detail::set_error_handler();
             detail::mpi_data_.error_handler_initialized_ = true;
         }
 
-        std::string name = get_pool_name();
-        // install polling loop on mpi thread pool
-        if (!detail::pool_exists_)
-        {
-            // drop back to default pika pool name
-            name = resource::get_pool_name(0);
-        }
-        // override mpi pool name with whatever we decided on
-        set_pool_name(name);
-        //
-        auto mode = mpi::experimental::get_completion_mode();
-        if (pika::mpi::experimental::detail::get_handler_mode(mode) !=
-            detail::handler_mode::yield_while)
-        {
-            PIKA_DETAIL_DP(detail::mpi_debug<5>,
-                debug(
-                    str<>("enabling polling"), name, mpi::experimental::detail::mode_string(mode)));
-            detail::register_polling(pika::resource::get_thread_pool(name));
-        }
-
         static bool defaults_set = false;
         if (!defaults_set)
         {
-            detail::init_throttling_default();
-            defaults_set = true;
+            // detail::init_throttling_default();
+            // defaults_set = true;
```

Comment:
What's this about?

Reply:
same as above. currently unused throttling code